operator/v1/types_ingress.go

```diff
@@ -632,6 +632,30 @@ const (
 	NeverHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Never"
 )
 
+// IngressControllerHTTPUniqueIdHeaderPolicy describes configuration for a
+// unique id header.
+type IngressControllerHTTPUniqueIdHeaderPolicy struct {
+	// name specifies the name of the HTTP header (for example, "unique-id")
+	// that the IngressController should inject into HTTP requests.  The
+	// field's value must be a valid HTTP header name as defined in RFC 2616
+	// section 4.2.  This field has no effect unless both name and format
+	// are specified.
+	//
+	// +optional
+	// +kubebuilder:validation:Pattern="^$|^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	Name string `json:"name,omitempty"`
+
+	// format specifies the format for the injected HTTP header's value.
```

Comment:
I woudl expect us to be able to choose a default format that will often work. Something that matches our UID format maybe?

Reply:
Added a default from the HAProxy documentation.